pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -29,9 +29,11 @@ def on_trainer_init(
         track_grad_norm: Union[int, float, str],
         accumulate_grad_batches: Union[int, Dict[int, int]],
         terminate_on_nan: bool,
+        detect_anomaly: bool,
     ):
 
         self.trainer.terminate_on_nan = terminate_on_nan
+        self.trainer.detect_anomaly = detect_anomaly
```

Comment:
we should set `terminate_on_nan` as well depending on detect_anomaly
e.g.
```
if terminate_on_nan is None:
    self.trainer.terminate_on_nan = detect_anomaly
else:
    # emit a deprecation warning
    self.trainer.terminate_on_nan = terminate_on_nan
```
a follow up can be to replace all usages of `trainer.terminate_on_nan` with `trainer.detect_anomaly`